src/main/webapp/utilities.js

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// A series of utility functions that will be helpful throughout the front-end
+
+/* eslint-disable no-unused-vars */
+
+/**
+ * Get cookie's value from document
+ * @param {string} cookieName the name of the cookie that needs to be found
+ * @return {string|null} either cookie's value or null if the cookie not found
+ */
+function getCookie(cookieName) {
+  // Cookie strings are in the format of
+  // cookieName1=cookieValue1; cookieName2=cookieValue2; ...
+  const allCookies = document.cookie.split('; ');
+
+  // Get the value of all matching cookies
+  const matchingCookies = allCookies.filter((cookie) => {
+    const retrievedCookieName = cookie.split('=')[0];
+    return retrievedCookieName === cookieName;
+  }).map((cookieString) => {
+    return cookieString.split('=')[1];
+  });
+
+  // If cookie is not found, or the name is duplicated, return null
+  if (matchingCookies.length !== 1) {
+    if (matchingCookies > 1) {
+      // This is unexpected behaviour
+      console.log('Duplicate cookie!');
+    }
+    return null;
+  }
+
+  return matchingCookies[0];
+}
+
+/**
+ * Checks if a cookie is present without returning its value
+ * @param {string} cookieName the name of the cookie that needs to be found
+ * @return {boolean} true if present, false otherwise
+ */
+function isCookiePresent(cookieName) {
+  return getCookie(cookieName) !== null;
+}
```

Comment:
I don't see this function being called yet. What use case should we anticipate for this?

Reply:
Will be used to check if authentication cookies are present. If they aren't, the site will have to prompt user to authenticate

Reply:
Ok, so this is to take care of cases like refreshing?

Reply:
This function itself won't handle the no authentication case - it will however enable a check which can be used to trigger a reauthentication. 

This is actually used in the next PR